lib/resty/shdict/simple.lua

```diff
@@ -54,7 +55,14 @@ function _M.gen_shdict_methods (opts)
                 end
 
                 -- with force:
-                ok, err = shdict:set(key, value, ttl)
+                local retries = 0
+                while not ok and force_retries > retries do
+                    ok, err = shdict:set(key, value, ttl)
+                    retries = retries + 1
+                    if DEBUG then
+                        dlog(ctx, 'try to set by force ' .. tostring(retries) .. 'th time')
+                    end
```

Comment:
We should only retry when the `err` is "no memory", not any other errors.

Reply:
fixed by https://github.com/openresty/lua-resty-shdict-simple/pull/4 @agentzh